File: 프로그래머스/2/49993. 스킬트리/스킬트리.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string skill, vector<string> skill_trees)
{
	int answer = 0;

	vector<char> temp;
	for (auto s : skill)
	{
		temp.push_back(s);
	}

	for (auto s : skill_trees)
	{
		int cnt = 0;
		bool chk1 = false;

		for (int i = 0; i < s.size(); i++)
		{
			if (cnt < temp.size() && temp[cnt] == s[i])
			{
				cnt++;
				continue;
			}
			for (int j = cnt+1; j < temp.size(); j++)
			{
				if (s[i] == temp[j])
				{
					chk1 = true;
					break;
				}
			}
			if (chk1)
			{
				break;
			}
		}
		if (chk1) continue;
		else answer++;


	}

	return answer;
}

File: 프로그래머스/2/49993. 스킬트리/스킬트리.cs
using System;

public class Solution {
    public int solution(string skill, string[] skill_trees) {
        int answer = 0;
        foreach(var item in skill_trees)
        {
            bool chk = true;
            int idx = 0;
            for(int i = 0; i<item.Length;i++)
            {
                if(idx >= skill.Length) break;
                if(item[i] == skill[idx])
                {
                    idx++;
                }
                else
                {
                    for(int j = idx+1; j < skill.Length;j++)
                    {
                        if(item[i] == skill[j])
                        {
                            chk = false;
                            break;
                        }
                    }
                }
            }
            if(chk) answer++;
        }
        return answer;
    }
}